src/utility_function_parameters.jl

```diff
@@ -0,0 +1,274 @@
+    log_weight_world = log(quantity_world) + (elasticity - 1.0) * (log_price_world - logP)
+
+    return log_weight_uk, log_weight_eu, log_weight_world
+
+end
+
+"""
+  Compute the total utility function parameter. Parameters of this type appear in multiple utility function
+  in the paper, and are annotated (at least) α, β1, β2, γ and ρ.
+
+  This is refactored from the Matlab code in e.g. ComputeTheta.m line 59
+
+  # Arguments
+  - log_price_index: price index computed by log_price_index()
+  - quantity: total quantity [f, x1, x2, I, m]
+  - elasticity: [ϵ, χ1, χ2, η, ξ]
+
+  # Outputs
+  - parameters
+"""
+function total_parameters(log_price_index::Vector{T}, quantity::Vector{T}, elasticity::T ) where {T <: Real}
+
+
+    length(log_price_index) == length(quantity) || error()
+
+    logPBar = log_total_price_index(elasticity, log_price_index, quantity)
+    parameters = Vector{T}(undef, length(log_price_index))
+
+    for i in axes(log_price_index,1)
+        parameters[i] = weight_kernel(quantity[i], exp(log_price_index[i] - logPBar), elasticity)
+    end
+
+    return parameters
+
+end
+
+"""
+  Compute the share of EU expenditure on UK exports, defined in section 2.2.1 of the main paper.
+"""
+function log_eu_expenditure_on_uk_exports(log_price_index::Vector{T}, quantity::Vector{T},
+                                          Ex::T, ETilde::T, ePx::T, PTilde::T, elasticity::T,
+                                          elasticity_tilde::T) where {T <: Real}
+
+    length(log_price_index) == length(quantity) || error()
+    logPBar = log_total_price_index(elasticity, log_price_index, quantity)
+
+    return log_weight_kernel(Ex/ETilde, exp(logPBar) * ePx / PTilde, elasticity_tilde)
+
+end
+
+"""
+  Compute the the consumer price index defined in equation 2.3 of the main paper as \bar{p}.
+  Matlab code reference e.g. ComputeTheta.m line 49.
+"""
+function price_index(elasticity::T,
+                     log_price_uk::T, log_price_eu::T, log_price_world::T,
+                     demand_uk::T, demand_eu::T, demand_world::T) where {T <: Real}
+
+    return (
+        demand_uk ^ (1.0/elasticity) * exp((elasticity - 1.0) * log_price_uk / elasticity) +
+        demand_eu ^ (1.0/elasticity) * exp((elasticity - 1.0) * log_price_eu / elasticity) +
+        demand_world  ^ (1.0/elasticity) * exp((elasticity - 1.0) * log_price_world  / elasticity)
+    ) ^ ( elasticity / (elasticity - 1.0) )
+
+end
+
+function log_price_index(elasticity::T,
+                         log_price_uk::T, log_price_eu::T, log_price_world::T,
+                         demand_uk::T, demand_eu::T, demand_world::T) where {T <: Real}
+
+    return elasticity / (elasticity - 1.0) * log(
+        demand_uk ^ (1.0/elasticity) * exp((elasticity - 1.0) * log_price_uk / elasticity) +
+        demand_eu ^ (1.0/elasticity) * exp((elasticity - 1.0) * log_price_eu / elasticity) +
+        demand_world  ^ (1.0/elasticity) * exp((elasticity - 1.0) * log_price_world  / elasticity)
+    )
```

Comment:
Minor nitpick: using `1` instead of `1.0` avoids promotion to `Float64` everywhere.  Probably not a concern right now, but in future that'd be useful if considering to use the same code on, say, GPUs where `Float32` would be preferable
```suggestion
    return elasticity / (elasticity - 1) * log(
        demand_uk ^ (1 / elasticity) * exp((elasticity - 1) * log_price_uk / elasticity) +
        demand_eu ^ (1 /elasticity) * exp((elasticity - 1) * log_price_eu / elasticity) +
        demand_world  ^ (1 / elasticity) * exp((elasticity - 1) * log_price_world  / elasticity)
    )
```